File: tests/go/cases/case_152_deferred_sink_gate_06.go
package escape_tests

var retainedCase152 = []map[string]string{}

func Case152DeferredSinkGate06(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "deferred_sink_gate_06",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "deferred_sink_gate_06:" + raw,
	}
	if len(raw) > 0 && raw[0] == 'x' {
		// ESCAPE: conditional path retention.
		retainedCase152 = append(retainedCase152, payload)
	}
	return "ok"
}

File: tests/nodejs/cases/case_152_deferred_sink_gate_06.js
/**
 * Task case 152: deferred_sink_gate_06 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case152DeferredSinkGate06(input) {
  const taskName = 'deferred_sink_gate_06';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  if (raw.startsWith('x')) {
    // ESCAPE: conditional sink only on selected path.
    retainedAudit.push(payload);
  }
  return 'ok';
}

module.exports = {
  case152DeferredSinkGate06,
};
